### PR TITLE
Updates to model  initialization methods

### PR DESCRIPTION
**Pull request by sourabhdalvi**

### CODE REVIEW

File: src/core/settings.jl [hunk 1]
```diff
@@ -7,6 +7,8 @@ struct Settings
     direct_mode_optimizer::Bool
     optimizer_log_print::Bool
     system_to_file::Bool
+    initialize_model::Bool
+    store_initial_conditions::Bool
```
Comment (jd-lara) [inline src/core/settings.jl:11]:
```suggestion
    serialize_initial_conditions::Bool
```

File: src/core/settings.jl [hunk 2]
```diff
@@ -42,6 +46,12 @@ function Settings(
         )
     end
 
+    if !initialize_model && store_initial_conditions
+        error(
```
Comment (jd-lara) [inline src/core/settings.jl:50]:
```suggestion
       throw(IS.ConflictingInputsError(
```

File: src/devices_models/device_constructors/thermalgeneration_constructor.jl [hunk 3]
```diff
@@ -1248,6 +1248,201 @@ function construct_device!(
...
 end
 
+function construct_device!(
+    container::OptimizationContainer,
+    sys::PSY.System,
+    ::ArgumentConstructStage,
+    model::DeviceModel{T, ThermalBasicCompactUnitCommitment},
+    ::Type{S},
+) where {T <: PSY.ThermalGen, S <: PM.AbstractPowerModel}
+    devices = PSY.get_components(T, sys)
```
Comment (jd-lara) [inline src/devices_models/device_constructors/thermalgeneration_constructor.jl:1258]: we should be using `get_available_components` here

File: src/devices_models/device_constructors/thermalgeneration_constructor.jl [hunk 4]
```diff
@@ -1248,6 +1248,201 @@ function construct_device!(
     return
 end
 
+function construct_device!(
```
Comment (jd-lara) [inline src/devices_models/device_constructors/thermalgeneration_constructor.jl:1251]: Why is this line not being covered in the tests?

File: src/operation/decision_model.jl [hunk 5]
```diff
@@ -48,6 +48,7 @@ mutable struct DecisionModel{M <: DecisionProblem} <: OperationModel
     name::Symbol
     template::ProblemTemplate
     sys::PSY.System
+    initialization_file::Union{AbstractString, Nothing}
```
Comment (jd-lara) [inline src/operation/decision_model.jl:51]:
```suggestion
    initialization_file::String
```

File: src/operation/decision_model.jl [hunk 6]
```diff
@@ -56,6 +57,7 @@ mutable struct DecisionModel{M <: DecisionProblem} <: OperationModel
         sys::PSY.System,
         settings::Settings,
         jump_model::Union{Nothing, JuMP.Model} = nothing;
+        initialization_file = nothing,
```
Comment (jd-lara) [inline src/operation/decision_model.jl:60]:
```suggestion
        initialization_file = "",
```

File: src/operation/decision_model.jl [hunk 7]
```diff
@@ -85,6 +87,9 @@ function DecisionModel{M}(
     horizon = UNSET_HORIZON,
     warm_start = true,
     system_to_file = true,
+    initialize_model = true,
+    initialization_file = nothing,
```
Comment (jd-lara) [inline src/operation/decision_model.jl:91]:
```suggestion
    initialization_file = "",
```

File: src/operation/decision_model.jl [hunk 8]
```diff
@@ -239,10 +253,13 @@ function build_pre_step!(model::DecisionModel)
         populate_contributing_devices!(get_template(model), get_system(model))
         add_services_to_device_model!(get_template(model))
 
-        @info "Make Initial Conditions  Model"
-        build_initial_conditions!(model)
-        initialize!(model)
-
+        run_initialization_step = get_initialize_model(get_settings(model))
+        initialization_file = get_initialization_file(model)
+        if run_initialization_step
```
Comment (jd-lara) [inline src/operation/decision_model.jl:258]:
```suggestion
        if run_initialization_step && isempty(initialization_file)
```

File: src/operation/decision_model.jl [hunk 9]
```diff
@@ -239,10 +253,13 @@ function build_pre_step!(model::DecisionModel)
...
+            @info "Make Initial Conditions  Model"
+            build_initial_conditions!(model)
+            initialize!(model)
```
Comment (jd-lara) [inline src/operation/decision_model.jl:261]: @daniel-thom here we would do something like this: 

```suggestion
            initialize!(model)
            elseif !(isempty(initialization_file)
                 # De-serialize initialization file
             else
                 error()
             end
```

File: src/operation/emulation_model.jl [hunk 10]
```diff
@@ -47,6 +47,7 @@ mutable struct EmulationModel{M <: EmulationProblem} <: OperationModel
     name::Symbol
     template::ProblemTemplate
     sys::PSY.System
+    initialization_file::Union{AbstractString, Nothing}
```
Comment (jd-lara) [inline src/operation/emulation_model.jl:50]:
```suggestion
    initialization_file::String
```

File: src/operation/emulation_model.jl [hunk 11]
```diff
@@ -56,6 +57,7 @@ mutable struct EmulationModel{M <: EmulationProblem} <: OperationModel
         sys::PSY.System,
         settings::Settings,
         jump_model::Union{Nothing, JuMP.Model} = nothing;
+        initialization_file = nothing,
```
Comment (jd-lara) [inline src/operation/emulation_model.jl:60]:
```suggestion
        initialization_file = "",
```

File: src/operation/emulation_model.jl [hunk 12]
```diff
@@ -91,6 +94,9 @@ function EmulationModel{M}(
     optimizer = nothing,
     warm_start = true,
     system_to_file = true,
+    initialize_model = true,
+    store_initial_conditions = false,
+    initialization_file = nothing,
```
Comment (jd-lara) [inline src/operation/emulation_model.jl:99]:
```suggestion
    initialization_file = "",
```

File: src/operation/emulation_model.jl [hunk 13]
```diff
@@ -246,9 +261,13 @@ function build_pre_step!(model::EmulationModel)
         populate_contributing_devices!(get_template(model), get_system(model))
         add_services_to_device_model!(get_template(model))
 
-        @info "Initial Conditions  Model"
-        build_initial_conditions!(model)
-        initialize!(model)
+        run_initialization_step = get_initialize_model(get_settings(model))
+        initialization_file = get_initialization_file(model)
+        if run_initialization_step
```
Comment (jd-lara) [inline src/operation/emulation_model.jl:266]: The same suggestions from above go here

**Review by jd-lara — CHANGES_REQUESTED** [review]

File: src/operation/operation_model_interface.jl [hunk 14]
```diff
@@ -127,7 +128,16 @@ function initialize!(model::OperationModel)
...
     @info "Solving Initialization Model for $(get_name(model))"
-    solve_impl!(model.internal.ic_model_container, get_system(model), Dict{Symbol, Any}())
+    try
+        solve_impl!(
+            model.internal.ic_model_container,
+            get_system(model),
+            Dict{Symbol, Any}(),
+        )
+    catch e
+        @error exception = (e, catch_backtrace())
```
Comment (daniel-thom) [inline src/operation/operation_model_interface.jl:138]: Can you add some text to this log message? such as `"initialize failed"`

File: src/core/settings.jl (hunk 2 above)
Comment (daniel-thom) [inline src/core/settings.jl:50]: You might want to use the IS.ConflictingInputsError exception here.

**Review by daniel-thom — APPROVED** [review]

**Review by jd-lara — APPROVED** [review]

**Pull request merged**